sdk/spring/scripts/replace_util.py

```diff
@@ -0,0 +1,199 @@
+    color_dict = {
+        'true': True,
+        'false': False
+    }
+    log.set_color(color_dict[args.color])
+
+
+def replace(module):
+    """
+    Replace action
+    :param module: module name
+    """
+    for pom in config[module].keys():
+        replace_artifact_id(module, pom)
+        replace_version(module, pom)
+
+
+def get_str(tuple_obj):
+    """
+    Return str list for tuple obj for logger.
+    :param tuple_obj: tuple obj
+    :return: string list
+    """
+    str_list = list()
+    for item in tuple_obj:
+        str_list.append(str(item))
+    return str_list
+
+
+def replace_artifact_id(module, pom):
+    """
+    Replace artifactId in dependency and plugin part.
+    :param module: module name
+    :param pom: pom file path
+    """
+    log.debug('Replacing artifact id in file: {}'.format(pom, module))
+    pom_dict = config[module][pom]
+    if not pom_dict.__contains__(ARTIFACT_ID_PAIRS_KEY_NAME):
+        log.warn('No config key {} in pom parameters.'.format(ARTIFACT_ID_PAIRS_KEY_NAME))
+        return
+
+    artifact_id_pairs = pom_dict[ARTIFACT_ID_PAIRS_KEY_NAME]
+    log.debug('Module: {}, artifact ids: {}'.format(module, get_str(artifact_id_pairs)))
+    with in_place.InPlace(pom) as file:
+        line_num = 0
+        for line in file:
+            line_num = line_num + 1
+            for artifact_id_pair in artifact_id_pairs:
+                new_line = line.replace(artifact_id_pair.old_artifact_id, artifact_id_pair.new_artifact_id)
+                if line != new_line:
+                    log.debug('Updating artifact id in line {}'.format(line_num))
+                    log.debug('    old_line = {}.'.format(line.strip('\n')))
+                    log.debug('    new_line = {}.'.format(new_line.strip('\n')))
+                    line = new_line
+            file.write(line)
+
+
+def replace_version(module, pom):
+    """
+    Replace version in dependency and plugin part.
+    :param module: module name
+    :param pom: pom file path
+    """
+    log.debug('Replacing version in file: {}'.format(pom))
+    pom_dict = config[module][pom]
+    if not pom_dict.__contains__(VERSION_UPDATE_ITEMS_KEY_NAME):
+        log.warn('No config key {} in pom parameters.'.format(VERSION_UPDATE_ITEMS_KEY_NAME))
+        return
+
+    version_update_items = pom_dict[VERSION_UPDATE_ITEMS_KEY_NAME]
+    log.debug('Module: {}, versions: {}'.format(module, get_str(version_update_items)))
+    with in_place.InPlace(pom) as file:
+        line_num = 0
+        for line in file:
+            line_num = line_num + 1
+            for version_update_item in version_update_items:
+                if version_update_item.id in line:
+                    log.debug('Updating line {}'.format(line_num))
+                    # update version in dependency part
+                    if X_VERSION_UPDATE in line:
+                        old_version = line[(line.index('<version>') + 9):line.index('</version>')]
```

Comment:
We can use 
```
if new_version != old_version
```
instead of 
```
if line != new_line:
```

Further more, if we configure property, old_version should not equal to new_version.
So `if check` can be deleted.

The code will be like this:

```
	                if X_VERSION_UPDATE in line:
                            old_version = line[(line.index('<version>') + 9):line.index('</version>')]
	                    new_line = line.replace(old_version, version_update_item.new_version)
                            log.debug('Updating version of dependency in line {}'.format(line_num))
                            log.debug('    old_line = {}.'.format(line.strip('\n')))
                            log.debug('    new_line = {}.'.format(new_line.strip('\n')))
                            line = new_line
```